compiler/rustc_middle/src/ty/context.rs

```diff
@@ -130,6 +130,31 @@ impl<'tcx> Interner for TyCtxt<'tcx> {
     ) -> (Self::Ty, ty::Mutability) {
         (ty, mutbl)
     }
+
+    fn mk_canonical_var_infos(self, infos: &[ty::CanonicalVarInfo<Self>]) -> Self::CanonicalVars {
+        self.mk_canonical_var_infos(infos)
+    }
+
+    fn mk_bound_ty(self, debruijn: ty::DebruijnIndex, var: ty::BoundVar) -> Self::Ty {
```

Comment:
either as a FIXME or implemented rn:

only add `mk_ty_from_kind` to interner and make `Ty::new_X` generic over the interner.